src/app/teams/teams.service.ts

```diff
@@ -15,9 +15,8 @@ export class TeamsService {
     this.teamsCollection = this.afs.collection('teams');
   }
 
-  addMembership(uid: string, displayName: string, email: string, teamId: string) {
-    this.teamsCollection.doc<Team>(teamId).valueChanges().subscribe(team => {
-      console.log(team.name);
+  addMembership(uid: string, displayName: string, email: string, teamId: string, emailService?: Function) {
```

Comment:
This isn't actually the service, just a function that sends the email, just badly named the variable. Will rename the variable to something like emailSender.

Reply:
nope, my point is to avoid passing in a function coming from join-team-page.component.ts
In this case what you can do instead is:

**join-team-page.component.ts:**
```
applyForTeam(user: User, teamId: string) {
  this.teamsService.addMembership(user.uid, user.displayName, user.email, teamId).pipe(
    flatMap(() => this.teamsService.getTeamLeadEmails(teamId))
  ).subscribe((teamLeadEmails: string[]) => {
    this.emailLeads(user.displayName, teamLeadEmails);
   // do something else, perhaps notify.update()?
  })
}
```
**teams.service.ts:**
```
getTeamLeadEmails(teamId: string): Observable<string[]> {
  const leadMembersCollection: AngularFirestoreCollection<Membership> = this.afs.collection('membership', ref =>
  ref.where('teamId', '==', teamId).where('isLead', '==', true));

  return leadMembersCollection.snapshotChanges().pipe(
    map((members) => members.map(item => item.payload.doc.data().email))
  );
}
```